### PR TITLE
App readers can log in to view article content

### PR DESCRIPTION
**Pull request by LucasKnudsen**
[PT Story](https://www.pivotaltracker.com/story/show/178319438)

## User story
```
As an App
In order to have premium functionality for our subscribers
I want only subscribers to be able to see article info (except for Featured)
```

## Changes proposed
* Adds authentication state
* Adds login view
* If not authenticated, user is routed to login view on article click
* Customizes stack back button

## What have we learned
* Customization of stack button
* Conditional routing

## Screenshot
image

### CODE REVIEW

File: components/Navigator.jsx [hunk 1]
```diff
@@ -1,58 +1,83 @@
...
+        />
+        <Stack.Screen
+          name='logIn'
+          component={LoginView}
+          options={({ navigation }) => ({
+            title: 'Back',
+            headerLeft: () => (
+              <AntDesign
+                name='arrowleft'
+                style={{ color: '#CEC269', paddingLeft: 15 }}
```
Comment (DavDan1) [inline components/Navigator.jsx:45]: I thought we put it in const now and group the styles, not inline :)

File: components/Navigator.jsx [hunk 2]
```diff
@@ -1,58 +1,83 @@
...
+            },
+          })}
+        />
+        <Stack.Screen
+          name='logIn'
+          component={LoginView}
+          options={({ navigation }) => ({
+            title: 'Back',
+            headerLeft: () => (
+              <AntDesign
```
Comment (offcast-dev) [inline components/Navigator.jsx:43]: didnt it had arrows there by default?

**Review by DavDan1 — APPROVED** [review]

File: views/LoginView.jsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  const [password, setPassword] = useState('');
+  const auth = new Authentication({ host: '' });
+  const article = route.params.article;
+
+  const authenticate = async () => {
+    try {
+      await auth.signIn(email, password);
+      navigation.navigate('single article', { article: article });
+    } catch (error) {
+      console.log(error);
```
Comment (offcast-dev) [inline views/LoginView.jsx]: what is the point of consolelog for user? user will never see it, I think you should delete it or display it on the page.

**Review by offcast-dev — COMMENTED** [review]
I would delete console.log

File: components/Navigator.jsx (hunk 2 above)
Comment (arunbhalli) [reply components/Navigator.jsx:43]: We customed the arrow on onpress functionality and it takes us directly to home page.

File: components/Navigator.jsx (hunk 1 above)
Comment (LucasKnudsen) [reply components/Navigator.jsx:45]: Well, both works awesomely!

File: views/LoginView.jsx (hunk 3 above)
Comment (LucasKnudsen) [reply views/LoginView.jsx]: IZ for all the hackers

**Review by christerforsgren91 — APPROVED** [review]

**Pull request merged**